x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/checkpoint/RemoteClusterResolver.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.transform.checkpoint;
+
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.transport.RemoteClusterAware;
+import org.elasticsearch.transport.RemoteConnectionStrategy;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CopyOnWriteArraySet;
+
+/**
+ * Maintain a list of remote clusters (aliases) and provide the ability to resolve.
+ */
+class RemoteClusterResolver extends RemoteClusterAware {
+
+    private final CopyOnWriteArraySet<String> clusters;
+
+    class ResolvedIndices {
+        private final Map<String, List<String>> remoteIndicesPerClusterAlias;
+        private final List<String> localIndices;
+
+        ResolvedIndices(Map<String, List<String>> remoteIndicesPerClusterAlias, List<String> localIndices) {
+            this.localIndices = localIndices;
+            this.remoteIndicesPerClusterAlias = remoteIndicesPerClusterAlias;
+        }
+
+        public Map<String, List<String>> getRemoteIndicesPerClusterAlias() {
+            return remoteIndicesPerClusterAlias;
+        }
+
+        public List<String> getLocalIndices() {
+            return localIndices;
+        }
+
+        public int size() {
```

Comment:
```suggestion
        public int numClusters() {
```

Or something. I thought `size()` was meant to be the number of total resolved indices, but instead it is used to help allocate the grouped listener.